src/api/app/models/concerns/workflow_run_gitea_payload.rb

```diff
@@ -10,6 +10,12 @@ def gitea_source_repository_full_name
     payload.dig(:repository, :full_name) # for a push event
   end
 
+  def gitea_target_repository_full_name
+    return payload.dig(:pull_request, :base, :repo, :full_name) if gitea_pull_request?
+
+    payload.dig(:repository, :full_name) # for a push event
```

Comment:
We should return nil if this isn't a push event...

Reply:
Fixed